### ISSUE TITLE
How do I look at the docker image that was built inside kaniko

### ISSUE THREAD

**Issue by rajakshay** [GoogleContainerTools/kaniko#740]
Hi,
I am trying to build a docker image using Kaniko. I am not sure if I am doing it right. I am using the `--no-push` option, since, I do not need to store that image anywhere. Doing so would be a waste of time and resources for our use-case. I have to use the image that I build in the next steps of my gitlab pipeline. 

The problem is that I am not sure if my image is being tagged correctly, and I am unable to refer to it in the pipeline, since I do not know what exactly the image name is.

I wish there was an option (something like `--overrideImageTag` to just override the tag for the image). There is no easy way to simulate `docker image tag` in kaniko when using the `--no-push` option. If there is, I am unaware.

I have tried the option `--tarPath`, but, when I extract the tar file, the content of the `manifest.json` is "null". I am probably doing something wrong. Need some guidance.

**Steps to re-produce the issue**
Start the kaniko docker container in interactive mode using the debug image.
```
docker run -it -v $(pwd):/workspace --entrypoint=/busybox/sh gcr.io/kaniko-project/executor:debug
```

**Once inside the kaniko container, you can create an image using:**
```
/kaniko/executor --dockerfile /workspace/Dockerfile --no-push --context /workspace --build-arg tag=app:app_version --tarPath=/workspace/image4.tar --cache=false
```

`ls -l workspace/image4.tar`
```
-rw-r--r--    1 0        0             2048 Aug 16 21:49 workspace/image4.tar
```
`cat manifest.json`
```null%```

**Image Build logs**
`/kaniko/executor --dockerfile /workspace/Dockerfile --no-push --context /workspace --build-arg tag=app:app_version --tarPath=/workspace/image4.tar --cache=false`
```
INFO[0000] Resolved base name hashicorp/terraform:0.12.3 to hashicorp/terraform:0.12.3
INFO[0000] Resolved base name hashicorp/terraform:0.12.3 to hashicorp/terraform:0.12.3
INFO[0000] Downloading base image hashicorp/terraform:0.12.3
INFO[0001] Error while retrieving image from cache: getting file info: stat /cache/sha256:66ac9a2d6e1dbe4edacc4d007a0ddcc5312482446f2ebcbe248e32b06a98e97e: no such file or directory
INFO[0001] Downloading base image hashicorp/terraform:0.12.3
INFO[0001] Built cross stage deps: map[]
INFO[0001] Downloading base image hashicorp/terraform:0.12.3
INFO[0002] Error while retrieving image from cache: getting file info: stat /cache/sha256:66ac9a2d6e1dbe4edacc4d007a0ddcc5312482446f2ebcbe248e32b06a98e97e: no such file or directory
INFO[0002] Downloading base image hashicorp/terraform:0.12.3
INFO[0002] Skipping unpacking as no commands require it.
INFO[0002] Taking snapshot of full filesystem...
INFO[0002] ENTRYPOINT ["/bin/sh"]
/ #
```

**Sample DockerFile**
```
FROM hashicorp/terraform:0.12.3
ENTRYPOINT ["/bin/sh"]
```

Please let me know if any other info would help.

**Comment by TimothySprague** [comment]
I had the same problem.  When I looked at the source code, I found that the `--tarPath` option is ultimately used [here](https://github.com/GoogleContainerTools/kaniko/blob/d67a8225031f2809f0d6c8c4f1825f233d8574a1/pkg/executor/push.go#L124).  Looking just above it's usage, I found that if no `--destination` options are found, then no image tags are actually generated.

To adjust your example image creation command:

```
/kaniko/executor --dockerfile /workspace/Dockerfile --no-push --context /workspace --destination app:app_version --tarPath=/workspace/image4.tar --cache=false
```
